l1-contracts/src.ts/hyperchain-upgrade.ts

```diff
@@ -11,55 +12,88 @@ import type { DiamondCut } from "./diamondCut";
 import { getFacetCutsForUpgrade } from "./diamondCut";
 
 import { getTokens } from "./deploy-token";
-import { EraLegacyChainId } from "./deploy";
 import type { Deployer } from "./deploy";
 
+import type { ITransparentUpgradeableProxy } from "../typechain/ITransparentUpgradeableProxy";
+import { ITransparentUpgradeableProxyFactory } from "../typechain/ITransparentUpgradeableProxyFactory";
+
 import { Interface } from "ethers/lib/utils";
+import { ADDRESS_ONE } from "./utils";
 import type { L2CanonicalTransaction, ProposedUpgrade, VerifierParams } from "./utils";
 
-import { REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT } from "zksync-web3/build/src/utils";
-import { ETH_ADDRESS_IN_CONTRACTS } from "zksync-ethers/build/src/utils";
+import {
+  REQUIRED_L2_GAS_PRICE_PER_PUBDATA,
+  priorityTxMaxGasLimit,
+  applyL1ToL2Alias,
+  hashL2Bytecode,
+} from "../../l2-contracts/src/utils";
 
 const SYSTEM_UPGRADE_TX_TYPE = 254;
 const FORCE_DEPLOYER_ADDRESS = "0x0000000000000000000000000000000000008007";
-// const CONTRACT_DEPLOYER_ADDRESS = "0x0000000000000000000000000000000000008006";
-// const COMPLEX_UPGRADE_ADDRESS = "0x000000000000000000000000000000000000800f";
 
-export async function upgradeToHyperchains(
+// const contractArtifactsPath = path.join("../../" as string, "contracts/l2-contracts/artifacts-zk/");
+// const l2BridgeArtifactsPath = path.join(contractArtifactsPath, "cache-zk/solpp-generated-contracts/bridge");
+// const openzeppelinBeaconProxyArtifactsPath = path.join(contractArtifactsPath, "@openzeppelin/contracts/proxy/beacon");
+// const systemContractsArtifactsPath = path.join("../.." as string, "contracts/system-contracts/??/");// kl todo
+
+// const L2_SHARED_BRIDGE_INTERFACE = readInterface(l2BridgeArtifactsPath, "L2SharedBridge");
+// const L2_SHARED_BRIDGE_IMPLEMENTATION_BYTECODE = readBytecode(l2BridgeArtifactsPath, "L2SharedBridge");
+const BEACON_PROXY_BYTECODE = ethers.constants.HashZero; //readBytecode(openzeppelinBeaconProxyArtifactsPath, "BeaconProxy");
+// const SYSTEM_CONTEXT_BYTECODE = readBytecode(systemContractsArtifactsPath, "SystemContext");
+
+export async function upgradeToHyperchains1(
```

Comment:
better to document what is the difference between `upgrade1` and `upgrade2` or choose better naming